src/simplnx/Filter/IFilter.cpp

```diff
@@ -219,20 +219,22 @@ IFilter::ExecuteResult IFilter::execute(DataStructure& data, const Arguments& ar
   // We can discard the warnings since they're already reported in preflight
   auto [resolvedArgs, warnings] = GetResolvedArgs(args, params, *this);
 
-  Result<> executeImplResult = executeImpl(data, resolvedArgs, pipelineFilter, messageHandler, shouldCancel);
+  Result<> executeImplResult = executeImpl(dataStructure, resolvedArgs, pipelineFilter, messageHandler, shouldCancel);
   if(shouldCancel)
   {
     return {MakeErrorResult(-1, "Filter cancelled")};
   }
 
-  Result<> preflightActionsExecuteResult = MergeResults(std::move(preflightActionsResult), std::move(executeImplResult));
+  Result<> validGeometries = MergeResults(dataStructure.validateGeometries(), executeImplResult);
+
```

Comment:
```suggestion
  Result<> validGeometries = MergeResults(dataStructure.validateGeometries(), executeImplResult);
  Result<> validMatrices = MergeResults(dataStructure.validateMatrices(), executeImplResult);

```
validateGeometries() does not finish validating the AttributeMatrix and completely misses any matrix created outside of a geometry's initialization (i.e. CreateAttributeMatrixFilter). Considering that it is the AttributeMatrix that was causing import errors, this is the part that needs attention to solve that specific bug.

This will require an additional method in DataStructure and AttributeMatrix.

Reply:
Done